client/rest/src/main/java/org/elasticsearch/client/RestClientBuilder.java

```diff
@@ -200,20 +202,25 @@ private CloseableHttpAsyncClient createHttpClient() {
             requestConfigBuilder = requestConfigCallback.customizeRequestConfig(requestConfigBuilder);
         }
 
-        HttpAsyncClientBuilder httpClientBuilder = HttpAsyncClientBuilder.create().setDefaultRequestConfig(requestConfigBuilder.build())
+        try {
+            HttpAsyncClientBuilder httpClientBuilder = HttpAsyncClientBuilder.create().setDefaultRequestConfig(requestConfigBuilder.build())
                 //default settings for connection pooling may be too constraining
-                .setMaxConnPerRoute(DEFAULT_MAX_CONN_PER_ROUTE).setMaxConnTotal(DEFAULT_MAX_CONN_TOTAL).useSystemProperties();
-        if (httpClientConfigCallback != null) {
-            httpClientBuilder = httpClientConfigCallback.customizeHttpClient(httpClientBuilder);
-        }
-
-        final HttpAsyncClientBuilder finalBuilder = httpClientBuilder;
-        return AccessController.doPrivileged(new PrivilegedAction<CloseableHttpAsyncClient>() {
-            @Override
-            public CloseableHttpAsyncClient run() {
-                return finalBuilder.build();
+                .setMaxConnPerRoute(DEFAULT_MAX_CONN_PER_ROUTE).setMaxConnTotal(DEFAULT_MAX_CONN_TOTAL)
+                .setSSLContext(SSLContext.getDefault());
```

Comment:
I just want to point out that this is slightly different from what is happening now. When it uses the system properties, Apache is calling `SSLContexts.createSystemDefault()`. Which does:

```
try {
    return SSLContext.getDefault();
} catch (final NoSuchAlgorithmException ex) {
    return createDefault();
}
```

And `createDefault()`:

```
try {
            final SSLContext sslcontext = SSLContext.getInstance(SSLContextBuilder.TLS);
            sslcontext.init(null, null, null);
            return sslcontext;
        } catch (final NoSuchAlgorithmException ex) {
            throw new SSLInitializationException(ex.getMessage(), ex);
        } catch (final KeyManagementException ex) {
            throw new SSLInitializationException(ex.getMessage(), ex);
        }
```

You know better than I if that change is okay. I just wanted to point it out.

Reply:
Thanks for looking into this. I was aware of it and the behavior change was intentional; I think the previous method was too lenient in that it silently ignored a bad parameter and swallowed the exception.